src/syscalls.c

```diff
@@ -1411,79 +1414,94 @@ bolos_err_t os_pki_get_info(uint8_t                  *key_usage,
+bolos_err_t ENDORSEMENT_get_public_key(ENDORSEMENT_slot_t slot,
+                                       uint8_t           *out_public_key,
+                                       uint8_t           *out_public_key_length)
 {
     unsigned int parameters[3];
-    parameters[0] = (unsigned int) index;
-    parameters[1] = (unsigned int) buffer;
-    parameters[2] = (unsigned int) length;
+    parameters[0] = (unsigned int) slot;
+    parameters[1] = (unsigned int) out_public_key;
+    parameters[2] = (unsigned int) out_public_key_length;
     return (unsigned int) SVC_Call(SYSCALL_os_endorsement_get_public_key_ID, parameters);
 }
 
-unsigned int os_endorsement_get_public_key_certificate(unsigned char  index,
-                                                       unsigned char *buffer,
-                                                       unsigned char *length)
+bolos_err_t ENDORSEMENT_get_public_key_certificate(ENDORSEMENT_slot_t endorsement_slot,
+                                                   uint8_t           *out_buffer,
+                                                   uint8_t           *out_length)
 {
     unsigned int parameters[3];
-    parameters[0] = (unsigned int) index;
-    parameters[1] = (unsigned int) buffer;
-    parameters[2] = (unsigned int) length;
+    parameters[0] = (unsigned int) endorsement_slot;
+    parameters[1] = (unsigned int) out_buffer;
+    parameters[2] = (unsigned int) out_length;
     return (unsigned int) SVC_Call(SYSCALL_os_endorsement_get_public_key_certificate_ID,
                                    parameters);
 }
 
-unsigned int os_endorsement_key1_get_app_secret(unsigned char *buffer)
+bolos_err_t ENDORSEMENT_key1_get_app_secret(uint8_t *out_secret)
 {
-    unsigned int parameters[2];
-    parameters[0] = (unsigned int) buffer;
-    parameters[1] = 0;
+    unsigned int parameters[1];
+    parameters[0] = (unsigned int) out_secret;
     return (unsigned int) SVC_Call(SYSCALL_os_endorsement_key1_get_app_secret_ID, parameters);
 }
 
-unsigned int os_endorsement_key1_sign_data(unsigned char *src,
-                                           unsigned int   srcLength,
-                                           unsigned char *signature)
+bolos_err_t ENDORSEMENT_key1_sign_data(uint8_t  *data,
+                                       uint32_t  data_length,
+                                       uint8_t  *out_signature,
+                                       uint32_t *out_signature_length)
 {
-    unsigned int parameters[3];
-    parameters[0] = (unsigned int) src;
-    parameters[1] = (unsigned int) srcLength;
-    parameters[2] = (unsigned int) signature;
+    unsigned int parameters[4];
+    parameters[0] = (unsigned int) data;
+    parameters[1] = data_length;
+    parameters[2] = (unsigned int) out_signature;
+    parameters[3] = (unsigned int) out_signature_length;
     return (unsigned int) SVC_Call(SYSCALL_os_endorsement_key1_sign_data_ID, parameters);
```

Comment:
I guess you also must update : SYSCALL_os_endorsement_key1_sign_data_ID from : 
0x03000059 to 0x04000059 because your are now returning 4 params ?